internal/controller/catalog/catalog_controller.go

```diff
@@ -318,6 +340,11 @@ func (r *CatalogReconciler) ensureKustomizationIsReady(ctx context.Context, cata
 	kustomization.SetNamespace(catalog.Namespace)
 	err = r.Get(ctx, client.ObjectKeyFromObject(kustomization), kustomization)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			requeue = true
+			err = nil
+			return
+		}
```

Comment:
These checks obfuscate the error. If the controller for some reason is not able to create the Kustomization resource it won't be visible on logs/catalog status etc.
Also what is the difference to just returning the error and let the machinery handle the requeueing?